frontend/src/types/analytics.ts

```diff
@@ -216,6 +216,18 @@ export type Events = {
     /** The permalink to the creator's profile */
     url: string
   }
+  /**
+   * Description: The user visits a source's link in the single result UI
```

Comment:
This event is for the external link to the source. Here, we can also get the id of the single result from which the link was clicked. However, for other source links (on the source collection page), there is no single media. Do we want to use the `id` as an optional prop, or is it not necessary?